### ISSUE TITLE
legacy checks

### ISSUE THREAD

**Issue by lucasjkr** [copona/copona#76]
There are lots of checks throughout opencart that we should be able to remove due to OC 2.3's upgraded system requirements, as well as our own additional requirement to use PHP 5.6.

For instance:

library/session.php has a create_id function returns different values based on the PHP version and PHP modules installed. Now that the mcrypt module is required to install Opencart, can we remove the checks for 'random_bytes' and 'openssl_random_pseudo_bytes'?

Likewise, the openbay module independently checks for the presence of mcrypt, which is not necessary since it's an opencart requirement.

Lots of general housekeeping like that. Any objection to getting that all cleaned up?

**Comment by arnisjuraga** [comment]
For sure! I am cleaning the code all the time! ;)
Just, please be careful. I have multiple sites, with different config and structure. Running PHP 5.6.9.

I don't have 5.6.0 PHP server, so if there are any very specific dependencies, we must be sure to test it.

But in the meantime - remove everything, just be careful :)

**Comment by lucasjkr** [comment]
Yes, I won't make changes based on PHP sub-versions, but only getting rid of things that are trully extraneous and just forgotten about, probably!

**Comment by lucasjkr** [comment]
OK, I just changed the version checks to PHP 5.6, but didn't touch any of the functions.

Speficially, for session Id generation, I'd rather use random_bytes, but that's 7.0 specific. There is a [package](https://github.com/paragonie/random_compat) that can be brought in through composer to provide that functionality to earlier versions, but that would make it more difficult to install for users who don't use composer or git. Also made a few other changes, pull request incoming.
